# Homework 4: Part 1

In [1]:
import numpy as np
import matplotlib.pyplot as plt
from scipy.integrate import odeint
import scipy.special as special
import astropy.units as u
import warnings
from scipy.integrate import quad
from scipy.optimize import fsolve
from sympy import *
import scipy.integrate as integ
%matplotlib inline
warnings.filterwarnings('ignore')

### Question 1: 

#### a)

$1+z = \frac{a(t_0)}{a(t_e)}$ where $a(t_0)$ is the scale factor when the light is observed, and $a(t_e)$ is the scale factor when the light is emitted. 

Therefore, $1 + z_1 = \frac{1}{a(t_1)}$
and 
$1 + z_2 = \frac{1}{a(t_2)}$

To get the redshift between the two galaxies, we will want to divide the two equations about so that we get $1 + Z = \frac{a(t_1)}{a(t_2)}$.

Doing this we get $1 + Z = \frac{1 + z_2}{1 + z_1}$, and therefore $Z = \frac{1 + z_2}{1 + z_1} - 1$

#### b)

To calculate the angular diameter distance we will need the equations:

$(\frac{H(z)}{H_0})^2 = \Omega_{r,0} \cdot (1+z)^4 + \Omega_{M,0} \cdot (1+z)^3 + \Omega_{\Lambda,0} + (1 - \Omega_0) \cdot (1+z)^2$ with $\Omega_{r,0}=0, \Omega_{\Lambda,0}=0, \Omega_{M,0}=1, \Omega_0=1$

$r(z) = c \cdot \int \frac{dz}{H(z)}$ for 0 to z.

and $d_A = \frac{r}{1 + z}$ for a flat universe.

In [2]:
def H(z):
    h = 70
    result = h*np.sqrt((1 + z)**3)
    return result

def r(z):
    c = 3*(10**5)
    result = c / H(z)
    return result

In [3]:
ans, err = quad(r, 0, .5) #r(z) for z_1 = .5 in Mpc
print ans

1572.88644919


In [4]:
dA1 = ans / (1+.5) #Angular diameter distance for z_1 = .5 in Mpc
print dA1

1048.59096613


$d_A = 1048.591$ Mpc for galaxy one with z = .5

In [5]:
ans2, err = quad(r, 0, 1.6) #r(z) for z_2 = 1.6 in Mpc
print ans2

3255.65423189


In [6]:
dA2 = ans2 / (1+1.6) #Angular diameter distance for z_2 = 1.6 in Mpc
print dA2

1252.17470457


$d_A = 1252.175$ Mpc for galaxy 2 with z = 1.6

In [7]:
ans3, err = quad(r, 0, .73) #r(z) for between galaxy 1 and 2 using Z which we calculated.
print ans3

2054.6921034


In [8]:
dA3 = ans3 / (1+.73) #Angular diameter distance for between galaxy 1 and 2, where z = z2 - z1
print dA3

1187.68329676


$d_A = 1187.863$ when looking at galaxy 2 from galaxy 1

#### c)

It makes sense that the angular diameter distance would be larger for galaxy 2 than galaxy 1 as it is farther away. From galaxy 1 we see galaxy 2 has a redshift of about .73, right in between the redshift for galaxy 1 and 2 when viewed from here. Therefore, it also makes sense that we got an angular diameter distance in between what we got for galaxy 1 with z = .5 and galaxy 2 with z = 1.6. 

### Question 2:

#### Ryden 8.5

Mean number density $n = \frac{N}{V}$ where N is the number of galaxies and V is the volume which in this case would be $\frac{4 \cdot \Pi}{3} \cdot (r_H)^3$. Therefore $n = \frac{3 \cdot N}{4 \cdot \Pi \cdot (r_H)^3}$

In [15]:
N = 1000
r = 1.5 #in Mpc
n = 3*N / (4*np.pi*r**3)
print n

70.7355302631


Therefore, $n = 70.74$ $galaxies \cdot Mpc^{-3}$

Next we want to know how far a galaxy in the Coma cluster will travel on average, before colliding with another galaxy. Here the cross section $\Sigma = 10^{-3} Mpc^2$.

In [26]:
E = 10**-3
d = 1 / (E*n) 
print d #distance in Mpc

14.1371669412


So on average, the distance a galaxy in the Coma cluster will travel before colliding with another galaxy would be 14.14 Mpc.

Lastly, we want to know what the average time between collisions is. For this we will need the velocity dispersion for the Coma cluster which is $\sigma = 880 km \cdot s^{-1}$. 

In [18]:
Mpc = 3.086*(10**19) #number of km in one Mpc
d2 = d*Mpc
print d2 #new value for d in km instead of Mpc

4.36272971804e+20


In [39]:
sigma = 880
t = d2 / (np.sqrt(3)*sigma) #multiplying sigma by sqrt(3) to get the square root of the mean square velocity.
print t #t in seconds

2.8622990649e+17


In [38]:
tyrs = t / (3.154*(10**7)) #converting seconds to years
print tyrs

9075139711.16


So the average time between collisions is approximately 9.08 Gyrs. This is smaller than the Hubble time of 13.7 Gyrs.